### PR TITLE
Clarify docs for sort(&mut self)

### PR DESCRIPTION
**Pull request by dns2utf8**
I documented the memory usage like noted in this issue:
- Document memory usage of sort() family of functions #33419

### CODE REVIEW

File: src/libcollections/slice.rs [hunk 1]
```diff
@@ -779,11 +779,11 @@ impl<T> [T] {
         core_slice::SliceExt::binary_search_by_key(self, b, f)
     }
 
-    /// Sorts the slice, in place.
-    ///
+
```
Comment (GuillaumeGomez) [inline src/libcollections/slice.rs]: Extra empty line.

File: src/libcollections/slice.rs [hunk 2]
```diff
@@ -779,11 +779,11 @@ impl<T> [T] {
...
     /// This is equivalent to `self.sort_by(|a, b| a.cmp(b))`.
     ///
-    /// This is a stable sort.
+    /// This sort is stable and `O(n log n)` worst-case, but allocates
```
Comment (GuillaumeGomez) [inline src/libcollections/slice.rs]: I think the comma is too much on this line.

File: src/libcollections/slice.rs [hunk 3]
```diff
@@ -804,11 +804,9 @@ impl<T> [T] {
     /// Sorts the slice, in place, using `key` to extract a key by which to
     /// order the sort by.
     ///
-    /// This sort is `O(n log n)` worst-case and stable, but allocates
+    /// This sort is stable and `O(n log n)` worst-case, but allocates
```
Comment (GuillaumeGomez) [inline src/libcollections/slice.rs]: I think the comma is too much on this line.

File: src/libcollections/slice.rs [hunk 4]
```diff
@@ -828,7 +826,7 @@ impl<T> [T] {
     /// Sorts the slice, in place, using `compare` to compare
     /// elements.
     ///
-    /// This sort is `O(n log n)` worst-case and stable, but allocates
+    /// This sort is stable and `O(n log n)` worst-case, but allocates
```
Comment (GuillaumeGomez) [inline src/libcollections/slice.rs]: I think the comma is too much on this line.

**Comment by steveklabnik** [comment]
I agree that the newline should be removed. The comma is fine, or fine to remove; I would probably write it, but I also know I tend towards using more commas than most people, and it reads fine either way.

**Comment by GuillaumeGomez** [comment]
It's just details, but in this case, I think that the comma is slowing down my reading needlessly.

**Comment by dns2utf8** [comment]
Hm, I like the commas, the seperate the sentences so well.

**Comment by dns2utf8** [comment]
Oh, I noticed I use a lot of them myself :smile: Removed them.

**Comment by dns2utf8** [comment]
Thank you too!

**Pull request merged**